Google.Api.Generator/Generation/MethodDetails.cs

```diff
@@ -275,6 +275,72 @@ private static MethodDetails DetectPagination(ServiceDetails svc, MethodDescript
                 field.IsRepeated && !field.IsMap && (field.FieldType == FieldType.Message || field.FieldType == FieldType.String);
         }
 
+        private static MethodDetails DetectRegapicPagination(ServiceDetails svc, MethodDescriptor desc)
+        {
+            var input = desc.InputType;
+            var output = desc.OutputType;
+
+            var resultsLimitCandidate = FindResultsLimitCandidate(input, IsResultsLimitCandidate);
+            var pageTokenCandidate = input.FindFieldByName("page_token");
+            var nextPageTokenCandidate = output.FindFieldByName("next_page_token");
+
+            var mapCandidateFields = output.Fields.InDeclarationOrder().Where(f => f.IsMap).ToList();
+            var repeatedCandidateFields = output.Fields.InDeclarationOrder().Where(IsRepeatedCandidate).ToList();
+            var repeatedField = mapCandidateFields.SingleOrDefault() ?? repeatedCandidateFields.SingleOrDefault();
```

Comment:
If there are multiple map fields or multiple repeated fields, I believe it's still valid so long as the first one numerically is also the first one in declaration order. At least, that's what the AIP says for regular pagination.

Also, if there *are* multiple fields, SingleOrDefault will give an exception which isn't terribly descriptive - I think it might be better to be explicit about how we handle multiple fields.

Reply:
👍